internal/controller/certificates/policies/checks_test.go

```diff
@@ -2277,3 +2277,64 @@ func Test_SecretOwnerReferenceMismatch(t *testing.T) {
 		})
 	}
 }
+
+func Test_SecretCertificateNameAnnotationsMismatch(t *testing.T) {
+	crt := gen.Certificate("test-certificate")
+
+	tests := map[string]struct {
+		input Input
+
+		expReason    string
+		expMessage   string
+		expViolation bool
+	}{
+		"without a CertificateName annotation, should return false": {
+			input: Input{
+				Certificate: crt,
+				Secret:      &corev1.Secret{},
+			},
+			expReason:    "",
+			expMessage:   "",
+			expViolation: false,
+		},
```

Comment:
Ok. The case I had in mind was where the Secret resource is pre-provisioned before the Certificate.
But I suppose in that case one of the other triggers would cause everything to be reconciled.